testsuite/tests/lib-format/tformat.ml

```diff
@@ -534,6 +536,545 @@ try
+  test (lasprintf "%Lu" [42L] = "42");
+  test (lasprintf "%-4Lu" [42L] = "42  ");
+  test (lasprintf "%04Lu" [42L] = "0042");
+  test (lasprintf "%+Lu" [42L] = "42");
+  test (lasprintf "% Lu" [42L] = "42");
+  test (lasprintf "%#Lu" [42L] = "42");
+  test (lasprintf "%4Lu" [42L] = "  42");
+  test (lasprintf "%*Lu" [4; 42L] = "  42");
+  test (lasprintf "%-0+ #6Ld" [42L] = "+42   ");
+
+  say "\nLu negative\n%!";
+  test (lasprintf "%Lu" [-1L] = "18446744073709551615");
+
+  say "\nLx positive\n%!";
+  test (lasprintf "%Lx" [42L] = "2a");
+  test (lasprintf "%-4Lx" [42L] = "2a  ");
+  test (lasprintf "%04Lx" [42L] = "002a");
+  test (lasprintf "%+Lx" [42L] = "2a");
+  test (lasprintf "% Lx" [42L] = "2a");
+  test (lasprintf "%#Lx" [42L] = "0x2a");
+  test (lasprintf "%4Lx" [42L] = "  2a");
+  test (lasprintf "%*Lx" [5; 42L] = "   2a");
+  test (lasprintf "%-0+ #*Lx" [5; 42L] = "0x2a ");
+
+  say "\nLx negative\n%!";
+  test (lasprintf "%Lx" [-42L] = "ffffffffffffffd6");
+
+  say "\nLX positive\n%!";
+  test (lasprintf "%LX" [42L] = "2A");
+  test (lasprintf "%-4LX" [42L] = "2A  ");
+  test (lasprintf "%04LX" [42L] = "002A");
+  test (lasprintf "%+LX" [42L] = "2A");
+  test (lasprintf "% LX" [42L] = "2A");
+  test (lasprintf "%#LX" [42L] = "0X2A");
+  test (lasprintf "%4LX" [42L] = "  2A");
+  test (lasprintf "%*LX" [5; 42L] = "   2A");
+  test (lasprintf "%-0+ #*LX" [5; 42L] = "0X2A ");
+
+  say "\nLX negative\n%!";
+  test (lasprintf "%LX" [-42L] = "FFFFFFFFFFFFFFD6");
+
+  say "\nLo positive\n%!";
+  test (lasprintf "%Lo" [42L] = "52");
+  test (lasprintf "%-4Lo" [42L] = "52  ");
+  test (lasprintf "%04Lo" [42L] = "0052");
+  test (lasprintf "%+Lo" [42L] = "52");
+  test (lasprintf "% Lo" [42L] = "52");
+  test (lasprintf "%#Lo" [42L] = "052");
+  test (lasprintf "%4Lo" [42L] = "  52");
+  test (lasprintf "%*Lo" [5; 42L] = "   52");
+  test (lasprintf "%-0+ #*Lo" [5; 42L] = "052  ");
+
+  say "\nLo negative\n%!";
+  test (lasprintf "%Lo" [-42L] = "1777777777777777777726");
+
+  say "\na\n%!";
+  let x = ref () in
+  let f y ppf _ = 
+    if y == !x then lfprintf ppf "ok" []
+    else lfprintf ppf "wrong" []
+  in
+  test (lasprintf "%a" [f (); x] = "ok");
+
+  say "\nt\n%!";
+  let f ppf = lfprintf ppf "ok" [] in
+  test (lasprintf "%t" [f] = "ok");
+
+  (* %{ fmt %} prints the signature of [fmt], i.e. a canonical representation
+    of the conversions present in [fmt].
+  *)
+  say "\n{...%%}\n%!";
+  let f = format_of_string "%f/%s" in
+  test (lasprintf "%{%f%s%}" [f] = "%f%s");
+
+  say "\n(...%%)\n%!";
+  let f = format_of_string "%d/foo/%s" in
+  test (lasprintf "%(%d%s%)" [f; 42; "bar"] = "42/foo/bar");
+
+  say "\n! %% @ , and constants\n%!";
+  test (lasprintf "%!" [] = "");
+  test (lasprintf "%%" [] = "%");
+  test (lasprintf "%@" [] = "@");
+  test (lasprintf "%," [] = "");
+  test (lasprintf "@@" [] = "@");
+  test (lasprintf "@@@@" [] = "@@");
+  test (lasprintf "@@%%" [] = "@%");
+
+  say "\nDelayed printf\n%!";
```

Comment:
Rather than duplicating existing tests on format specifiers, I think it would be clearer for both printers and format to have a new `lprintf` tests with only the miscellaneous and delayed section.

Reply:
The latest commit splits the het-list tests into new files for both `Format` and `Printf`